build-tools/packages/build-tools/src/type-test-generator/typeTestUtils.ts

```diff
@@ -0,0 +1,289 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { readJsonSync } from "fs-extra";
+import path from "node:path";
+import { ExtractorConfig } from "@microsoft/api-extractor";
+import { existsSync, rmSync } from "node:fs";
+import { Project, SourceFile } from "ts-morph";
+import { BrokenCompatTypes } from "../common/fluidRepo";
+import { buildTestCase, TestCaseTypeData } from "../typeValidator/testGeneration";
+import { getFullTypeName, getNodeTypeData, TypeData } from "../typeValidator/typeData";
+import { PackageJson } from "../common/npmPackage";
+
+// Information about the previous package from the package.json is not needed,
+// but error if its missing since it's nice to separate errors for the dep missing here vs not installed.
+// This ensures that a critical dependency (the previous package version) is correctly declared in the project's package.json.
+export function ensureDevDependencyExists(
+	packageObject: PackageJson,
+	dependencyName: string,
+): void {
+	const dependencyVersion = packageObject?.devDependencies?.[dependencyName];
+	if (typeof dependencyVersion !== "string") {
+		throw new Error(`Did not find devDependency ${dependencyName} in package.json`);
+	}
+}
+
+/**
+ * Fetches the path of the previous package.json or throws an error if not found.
+ * @param previousBasePath - A string representing the previous package.json path
+ * @returns
+ */
+export function getPreviousPackageJsonPath(previousBasePath: string): string {
+	const previousPackageJsonPath = `${previousBasePath}/package.json`;
+	if (!existsSync(previousPackageJsonPath)) {
+		throw new Error(`${previousPackageJsonPath} not found.`);
+	}
+	return previousPackageJsonPath;
+}
+
+/**
+ * Attempts to retrieve  a specified type of rollup file path for type definitions from the API Extractor configuration.
+ * @param {string} rollupType - The type of rollup file path to retrieve (ex: "alpha", "beta", "public").
+ * @returns {string} The path to the alpha trimmed type definitions file.
+ * @throws {Error} If api-extractor config cannot be loaded or if the alpha trimmed file path is undefined
```

Comment:
The comments currently don't match the implementation (it never throws), but I think what the new comments indicate is acceptable behavior. If we cannot find the api-extractor config this method should throw. I think we only do type tests for packages where we have this config, it's kind of necessary. If the config exists but there's no file for the given rollupType, then return undefined.
```suggestion
 * @returns {string} The path to the type definitions file for the specified rollupType, or undefined it cannot be found.
 * @throws {Error} If api-extractor config cannot be loaded.
```

When handling unexpected errors, inside the catch we log to `console.error()` so we have a chance to say "something bad occurred while attempting X" but then do `throw error;` (the same error in the `catch(error)` so it preserves its stack trace).